parser-library/buffer.cpp

```diff
@@ -291,16 +301,11 @@ void deleteBuffer(bounded_buffer *b) {
 #endif
   }
 
-  if (b->detail != nullptr) {
```

Comment:
Deleting a pointer set to nullptr has no UB and is supported by the standard, so I removed the if.